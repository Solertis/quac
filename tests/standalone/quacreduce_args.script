#!/bin/bash

# Test that quacreduce argument parsing works correctly.
#
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR

set +e
exec 2>&1  # send stderr to stdout to make output more readable
#set -x    # safe because we have no pipelines in this script


echo '*** Check input files for unique names'
x quacreduce --map cat --reduce cat foo/bar.txt baz/qux.txt
echo $?
x quacreduce --map cat --reduce cat foo/bar.txt baz/bar.txt
echo $?

echo '*** Check that either --python xor --map and --reduce are specified'

echo '*** these should succeed'
#x quacreduce --python foo job1 input.txt
#echo $?
x quacreduce --map foo --reduce bar input.txt
echo $?

echo '*** these should fail'
x quacreduce --notimes input.txt
echo $?
x quacreduce --map bar input.txt
echo $?
x quacreduce --reduce bar input.txt
echo $?
x quacreduce --python foo --map bar input.txt
echo $?
x quacreduce --python foo --reduce bar input.txt
echo $?
x quacreduce --python foo --map bar --reduce baz input.txt
echo $?

echo '*** Check that job fails if the map or reduce commands fail'
x quacreduce --notimes --map false --reduce true /dev/null
y "make || true"
x quacreduce --notimes --map true --reduce false /dev/null
y "make || true"
