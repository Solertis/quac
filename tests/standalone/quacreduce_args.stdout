*** Check input files for unique names
$ quacreduce --map cat --reduce cat foo/bar.txt baz/qux.txt
0
$ quacreduce --map cat --reduce cat foo/bar.txt baz/bar.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--pyargs DICT]
                  [--jobdir DIR] [--partitions N] [--sortdir DIR]
                  [--sortmem N] [--update] [-h] [--notimes] [--unittest]
                  [--verbose]
                  FILE [FILE ...]
quacreduce: error: input file basenames must be unique
2
*** Check that either --python xor --map and --reduce are specified
*** these should succeed
$ quacreduce --map foo --reduce bar input.txt
0
*** these should fail
$ quacreduce --notimes input.txt
quacr FATAL    --python or --map and --reduce must be specified
1
$ quacreduce --map bar input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--pyargs DICT]
                  [--jobdir DIR] [--partitions N] [--sortdir DIR]
                  [--sortmem N] [--update] [-h] [--notimes] [--unittest]
                  [--verbose]
                  FILE [FILE ...]
quacreduce: error: --map and --reduce must both be specified if one is
2
$ quacreduce --reduce bar input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--pyargs DICT]
                  [--jobdir DIR] [--partitions N] [--sortdir DIR]
                  [--sortmem N] [--update] [-h] [--notimes] [--unittest]
                  [--verbose]
                  FILE [FILE ...]
quacreduce: error: --map and --reduce must both be specified if one is
2
$ quacreduce --python foo --map bar input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--pyargs DICT]
                  [--jobdir DIR] [--partitions N] [--sortdir DIR]
                  [--sortmem N] [--update] [-h] [--notimes] [--unittest]
                  [--verbose]
                  FILE [FILE ...]
quacreduce: error: --python cannot be specified with --map and/or --reduce
2
$ quacreduce --python foo --reduce bar input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--pyargs DICT]
                  [--jobdir DIR] [--partitions N] [--sortdir DIR]
                  [--sortmem N] [--update] [-h] [--notimes] [--unittest]
                  [--verbose]
                  FILE [FILE ...]
quacreduce: error: --python cannot be specified with --map and/or --reduce
2
$ quacreduce --python foo --map bar --reduce baz input.txt
usage: quacreduce [--map CMD] [--reduce CMD] [--python CLASS] [--pyargs DICT]
                  [--jobdir DIR] [--partitions N] [--sortdir DIR]
                  [--sortmem N] [--update] [-h] [--notimes] [--unittest]
                  [--verbose]
                  FILE [FILE ...]
quacreduce: error: --python cannot be specified with --map and/or --reduce
2
*** Check that job fails if the map or reduce commands fail
$ quacreduce --notimes --map false --reduce true /dev/null
$ (make || true)
false < /dev/null | hashsplit 1 tmp/null && if [ ${PIPESTATUS[0]} -ne 0 ]; then false; fi
make: *** [tmp/null.mapped] Error 1
$ quacreduce --notimes --map true --reduce false /dev/null
$ (make || true)
true < /dev/null | hashsplit 1 tmp/null && if [ ${PIPESTATUS[0]} -ne 0 ]; then false; fi
touch tmp/null.mapped
LC_ALL=C sort -s -k1,1 -t'	' -S 64M -T tmp tmp/null.0 | false && if [ ${PIPESTATUS[0]} -ne 0 ]; then false; fi
make: *** [tmp/0.reduced] Error 1
